File: bestTimeToBuyAndSellStock.js
var maxProfit = function (price) {
    var minPrice = price[0];
    var maxProfit = 0;
    var n = price.length;

    for (var i = 0; i < n; i++) {
        if (minPrice > price[i]) {
            minPrice = price[i];
        } else {
            var profit = price[i] - minPrice;
            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }
    }

    if (maxProfit > 0) {
        return maxProfit;
    } else {
        return 0;
    }
}

File: bestTimeToBuyAndSellStock.c
#include <stdio.h>

int main()
{
    int n, i;
    printf("Enter number of days: ");
    scanf("%d", &n);

    int price[n];
    printf("Enter prices: ");
    for (i = 0; i < n; i++)
    {
        scanf("%d", &price[i]);
    }

    int minPrice = price[0];
    int maxProfit = 0;

    for (i = 0; i < n; i++)
    {
        if (price[i] < minPrice)
        {
            minPrice = price[i];
        }
        else
        {
            int profit = price[i] - minPrice;
            if (profit > maxProfit)
            {
                maxProfit = profit;
            }
        }
    }

    if (maxProfit > 0)
        printf("Maximum Profit = %d\n", maxProfit);
    else
        printf("Maximum Profit = 0\n");

    return 0;
}
